packages/casl-mongoose/src/accessible_fields.ts

```diff
@@ -8,7 +8,10 @@ export type AccessibleFieldsOptions =
   } &
   ({ only: string | string[] } | { except: string | string[] });
 
-export const getSchemaPaths: AccessibleFieldsOptions['getFields'] = schema => Object.keys((schema as { paths: object }).paths);
+export const getSchemaPaths: AccessibleFieldsOptions['getFields'] = schema => [
+  ...Object.keys((schema as { paths: object }).paths),
+  ...Object.keys((schema as { virtuals: object }).virtuals)
+];
```

Comment:
**issue(blocking):** this is a breaking change. And I don't think that everybody needs/wants virtual fields to appear in accessibleFields, especially that some of them may be non-writab

Reply:
> after a 2nd look at the code I don't understand why you can't apply plugin globally with custom `getFields` option, did I miss smth?
> 
> ```ts
> mongoose.plugin(accessibleFieldsPlugin, {
>   getFields: schema => [...Object.keys(schema.paths), ...Object.keys(schema.virtuals)]
> });
> ```

After looking at it again i was able to make it work globally, i don't know why it didn't work the first time, guess i did something wrong. It's a bit confusing some times when working with typescript.

And yeah, it's a breaking change and if it works globally with a few lines i don't think it's worth it. I'll better close this PR.
It was all my misunderstanding, sorry